(DOCSP-26537): Updating list of Kotlin-supported data types (#2384)
mongodb/docs-realm: source/sdk/kotlin/realm-database/schemas/supported-types.txt

- ``Char``
- ``Short``
- ``Int``
- ``MutableRealmInt``, which behaves like a ``Long`` but also supports ``increment`` 
  and ``decrement`` methods that implement a conflict-free replicated data type.  
  For more information, see `MutableRealmInt <{+kotlin-local-prefix+}io.realm.kotlin.types/-mutable-realm-int/index.html>`__.   
- ``Long``
- ``Boolean``
- ``Double``
- ``Float``
- ``ObjectId`` 
- ``RealmInstant``
- Any ``RealmObject`` subclass
- ``RealmList<T>``, where T is any of the supported data types or a   
  `RealmObject <{+kotlin-local-prefix+}io.realm.kotlin.types/-realm-object/index.html>`__.

>   Unless T is a ``RealmObject``, T can also be nullable (``RealmList<T?>``).

Reviewer: Suggestion, I think the following would read better (but you can take this advice or leave it):

```suggestion
   Lists of ``RealmObject`` cannot have null elements. All other types of ``RealmList<T>`` can be nullable (``RealmList<T?>``).
```

Author: Reworded

- ``RealmSet<T>``, where T is any of the supported data types or a
  `RealmObject <{+kotlin-local-prefix+}io.realm.kotlin.types/-realm-object/index.html>`__.

>   Unless T is a ``RealmObject``, T can also be nullable (``RealmSet<T?>``).

Reviewer: Similar suggestion as above: https://github.com/mongodb/docs-realm/pull/2384/files#r1044837668

- ``RealmUUID``

> - ``BacklinksDelegate<T>``, which is used to define an inverse relationship between 
>   `RealmObjects <{+kotlin-local-prefix+}io.realm.kotlin.types/-realm-object/index.html>`__ using 
>   `backlinks <{+kotlin-local-prefix+}io.realm.kotlin.ext/backlinks.html>`__.

Reviewer 2: ```suggestion
- ``BacklinksDelegate<T>``, a `backlinks <{+kotlin-local-prefix+}io.realm.kotlin.ext/backlinks.html>` delegate used to define an inverse relationship between 
  `RealmObjects <{+kotlin-local-prefix+}io.realm.kotlin.types/-realm-object/index.html>`__. 
```

While technically correct, most people will not really understand it. The reference to the backlink docs are great, but I tried to reword a little to put more emphasis on it. Not 100% sure I got the formatting correct.
